app/models/manageiq/providers/inventory.rb

```diff
@@ -45,34 +47,47 @@ def inventory_collections
       parse.inventory_collections
     end
 
+    # Based on the given provider/manager class, this returns correct collector class
+    #
+    # @param target class of the Provider/Manager
+    # @return [Class] Correct class name of the collector
+    def self.collector_class_for(ems, target = nil)
+      target = ems if target.nil?
+      class_for(ems, target, 'Collector')
+    end
+
     # Based on the given provider/manager class, this returns correct persister class
     #
-    # @param klass class of the Provider/Manager
+    # @param ems class of the Provider/Manager
+    # @param target class of refresh's target
     # @return [Class] Correct class name of the persister
-    def self.persister_class_for(klass)
-      provider_module = ManageIQ::Providers::Inflector.provider_module(klass)
-      "#{provider_module}::Inventory::Persister::#{klass.name.demodulize}".safe_constantize
-    rescue ManageIQ::Providers::Inflector::ObjectNotNamespacedError => _err
-      nil
+    def self.persister_class_for(ems, target = nil)
+      target = ems if target.nil?
```

Comment:
there is a need for one-param method from there: https://github.com/ManageIQ/manageiq/blob/master/app/models/ems_refresh.rb#L133